### DISCUSSION TITLE
mix pink and white noise / low- band and high pass filter / volume control

### DISCUSSION THREAD

**Discussion by helste64** [pschatzmann/arduino-audio-tools#328] [Q&A] [answered]
I have realized a project with Teensy 4.0 and Teensy Audio Shield (Audio System Design Tool - very cool), where two external signals (Pink and White noise) are mixed.
Furthermore a low-, band- and high-pass filter and a volume control are used to adjust the sound / volume.
The whole is controlled via a Bluetooth module (HM10 or HM18).
If the power supply fails, then all values (low, band high-pass filter and volume) are stored via eeprom, so that these settings are not lost.
Now my request / question before I read into the matter (Audio-Kit + Audio Tools ) - can you evaluate whether this project can also be implemented with your super audio kit + audio tools?

**Comment by pschatzmann** [comment, marked as answer]
I would tend to think probably yes. My understanding is that the major drawback of the ESP32 is that the floating point operations are much slower then on the Teensy. So you should investigate the performance of the filters first. My filter implementation is platform independent, so it is a kind of slow, but Espressif has some optimized assembler libraries.
